### PR TITLE
feat(claim): Add SYCNED condition to XRs and XRCs

### LINKED ISSUE #2850: Add SYNCED status column to XRC and XR
### What problem are you facing?

Currently Crossplane adds a `READY` column to a XRC and XR that is either `True` or `False` depending on the state of the managed resources:

```yaml
- additionalPrinterColumns:
    - jsonPath: .status.conditions[?(@.type=='Ready')].status
      name: READY
      type: string
```

However, it is difficult to distinguish between MR and composition issues without looking inside the XRC or XR. This can be especially troublesome if you have a lot of XRCs.

### How could Crossplane help solve your problem?

Crossplane should add an additional column called `SYNCED` to every XRC and XR similar to the `SYNCED` column of managed resources:

```yaml
    - jsonPath: .status.conditions[?(@.type=='Synced')].status
      name: SYNCED
      type: string
```

This would require adding an additional `Synced` condition to `status.conditions`, i.e:

```yaml
  - lastTransitionTime: "2021-12-17T12:50:27Z"
    reason: ReconcileSuccess
    status: "True"
    type: Synced
```

### PR DESCRIPTION
**Pull request by MisterMX**
### Description of your changes

Fixes #2850

* Add `Synced` condition that can be used to track reconcile errors
(similar to managed resources in providers)
* Add an additional printer column to XR CRDs that references the
`Synced` condition

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Manually and unit tests

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/apiextensions/claim/reconciler.go
Comment (negz) [inline internal/controller/apiextensions/claim/reconciler.go:305]: I'd prefer to update the reconciler to set `ReconcileError` as part of the existing `SetConditions` calls, rather than moving the bulk of the logic into a wrapped `reconcile` method. The former approach is more in line with how we approach this in the managed reconciler, for example:

https://github.com/crossplane/crossplane-runtime/blob/988c9ba9c255d7cade32e2c0135ce47b083eaf95/pkg/reconciler/managed/reconciler.go#L684

**Review by negz — APPROVED** [review]

**Comment by negz** [comment]
Apparently I can't merge this PR - I think because it hasn't been approved by any CODEOWNERS (which would in this case be a member of the reviewers group or muvaf).

**Review by hasheddan — APPROVED** [review]
Stamping to unblock merge based on @negz approval.

**Review by bobh66 — APPROVED** [review]

**Pull request merged**
